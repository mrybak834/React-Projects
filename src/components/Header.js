import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startSignOut } from '../redux/actions/auth';

export const Header = (props) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/dashboard' activeClassName='is-active' exact={true}>Dashboard </NavLink>
        <NavLink to='/new' activeClassName='is-active' exact={true}>New </NavLink>
        <button onClick={props.startSignOut}>Sign Out</button>
        <br/>
        ------------------------------
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startSignOut: () => dispatch(startSignOut())
});

export default connect(undefined, mapDispatchToProps)(Header);
