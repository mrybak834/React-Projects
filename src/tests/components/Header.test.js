/**
 * Shallow rendering renders only the given component
 * as opposed to full DOM rendering which renders child components
 */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';




test('Components: Header: Render Header', () => {
    const wrapper = shallow(<Header />);

    /**
     * Snapshot assertions only work with multiple runs.
     * First run creates a snapshot, all subsequent runs must match the original snapshot
     * Wrappers are serialized and stripped via the serializers in jest.config
     */
    expect(wrapper).toMatchSnapshot();

});