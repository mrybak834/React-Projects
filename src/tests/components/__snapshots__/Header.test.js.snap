// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Header: Render Header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/dashboard"
  >
    Dashboard 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/new"
  >
    New 
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    Sign Out
  </button>
  <br />
  ------------------------------
</header>
`;
